src/parser/parser_support.c

```diff
@@ -9887,3 +9898,258 @@ pt_has_name_oid (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *contin
+  const char *original_name = NULL;
+  const char *resolved_name = NULL;
+  char downcase_resolved_name[DB_MAX_USER_LENGTH] = { '\0' };
+  char *current_user_name = NULL;
+  const char *user_specified_name = NULL;
+  int error = NO_ERROR;
+
+  if (!node)
+    {
+      return NULL;
+    }
+
+  if (PT_IS_NAME_NODE (node) && PT_NAME_INFO_IS_FLAGED (node, PT_NAME_INFO_USER_SPECIFIED))
+    {
+      original_name = node->info.name.original;
+      resolved_name = node->info.name.resolved;
+    }
+  else if (PT_IS_EXPR_NODE (node) && PT_IS_SERIAL (node->info.expr.op))
+    {
+      if (PT_IS_DOT_NODE (node->info.expr.arg1)
+	  && PT_IS_NAME_NODE (node->info.expr.arg1->info.dot.arg1)
+	  && PT_IS_NAME_NODE (node->info.expr.arg1->info.dot.arg2))
+	{
+	  PT_NODE *owner = node->info.expr.arg1->info.dot.arg1;
+	  PT_NODE *name = node->info.expr.arg1->info.dot.arg2;
+
+	  original_name = name->info.name.original;
+	  resolved_name = owner->info.name.original;
+	}
+      else
+	{
+	  assert (PT_IS_NAME_NODE (node->info.expr.arg1));
+
+	  PT_NODE *name = node->info.expr.arg1;
+
+	  original_name = name->info.name.original;
+	  resolved_name = name->info.name.resolved;
+	}
+    }
+  else
+    {
+      return node;
+    }
+
+  assert (original_name && original_name[0] != '\0');
+
+  if (strchr (original_name, '.'))
+    {
+      /* It is already user_specified_name. */
+      return node;
+    }
+
+  /* In the system_class, user_specified_name does not include user_name.
+   * So, info.name.original is different for each case below.
+   *
+   *    original_name        resolve_name        user_specified_name
+   * -------------------------------------------------------------------------------
+   * 1. common_class_name &&             NULL -> current_user_name.common_class_name
+   * 2. common_class_name && common_user_name ->  common_user_name.common_class_name
+   * 3. common_class_name &&    dba_user_name ->     dba_user_name.common_class_name
+   * 4. system_class_name &&             NULL ->                   system_class_name
+   * 5. system_class_name && common_user_name ->  common_user_name.system_class_name
+   * 6. system_class_name &&    dba_user_name ->                   system_class_name
+   * 
+   * In case 5, raises an error to inform the user of an incorrect customization.
+   */
+  if (!PT_IS_SERIAL (node->info.expr.op) && sm_check_system_class_by_name (original_name))
+    {
+      /* Skip in case 4, 6 */
+      if (resolved_name == NULL || resolved_name[0] == '\0' || intl_identifier_casecmp (resolved_name, "DBA") == 0)
+	{
+	  return node;
+	}
+    }
+
+  if (resolved_name == NULL || resolved_name[0] == '\0')
+    {
+      current_user_name = db_get_user_name ();
+      if (!current_user_name)
+	{
+	  ASSERT_ERROR_AND_SET (error);
+	  PT_ERROR (parser, node, er_msg ());
+	  *continue_walk = PT_STOP_WALK;
+	  return node;
+	}
+
+      resolved_name = current_user_name;
+    }
+
+  if (strlen (resolved_name) >= DB_MAX_USER_LENGTH)
```

Comment:
I don't think the user name obtained through db_get_user_name() can exceed the length limit.
So how about doing something like below?
`else if (strlen (resolved_name) >= DB_MAX_USER_LENGTH)`

Reply:
yes!